### PR TITLE
[Merged by Bors] - feat(Normed/Group): add `norm_abs_zsmul` and `norm_neg_one_pow_zsmul`

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Normed/Group/Basic.lean [hunk 1]
```diff
@@ -375,6 +375,14 @@ theorem norm_div_rev (a b : E) : ‖a / b‖ = ‖b / a‖ := by
 @[to_additive (attr := simp) norm_neg]
 theorem norm_inv' (a : E) : ‖a⁻¹‖ = ‖a‖ := by simpa using norm_div_rev 1 a
 
+@[to_additive (attr := simp) norm_abs_zsmul]
+theorem norm_zpow_abs (a : E) (n : ℤ) : ‖a ^ |n|‖ = ‖a ^ n‖ := by
+  rcases le_total 0 n with hn | hn <;> simp [hn, abs_of_nonneg, abs_of_nonpos]
+
+@[to_additive (attr := simp) norm_neg_one_pow_zsmul]
+theorem norm_zpow_neg_one_pow (a : E) (n : ℕ) : ‖a ^ ((-1) ^ n : ℤ)‖ = ‖a‖ := by
```
Comment (urkud) [inline Mathlib/Analysis/Normed/Group/Basic.lean:383]: Is this one too specific?

Comment (eric-wieser) [reply Mathlib/Analysis/Normed/Group/Basic.lean:383]: How about
```lean
theorem norm_zpow_of_is_unit (a : E) (z : ℤ) (hz : IsUnit z) : ‖a ^ z‖ = ‖a‖ := by
```
instead, from which this follows trivially?

Comment (eric-wieser) [reply Mathlib/Analysis/Normed/Group/Basic.lean:383]: (this also works nicely if working with `Units ℤ` since you can then use `u.isUnit`)

Comment (urkud) [reply Mathlib/Analysis/Normed/Group/Basic.lean]: We don't have `Pow _ (Units _)`, so I added an additive-only lemma for `SMul (Units Int) E`.

**Review by eric-wieser — APPROVED** [review]
bors d+

The nnnorm versions would be nice to have too

File: Mathlib/Analysis/Normed/Group/Basic.lean [hunk 2]
```diff
@@ -375,6 +375,22 @@ theorem norm_div_rev (a b : E) : ‖a / b‖ = ‖b / a‖ := by
...
+@[to_additive (attr := simp) norm_natAbs_smul]
+theorem norm_pow_natAbs (a : E) (n : ℤ) : ‖a ^ n.natAbs‖ = ‖a ^ n‖ := by
+  rw [← zpow_natCast, ← Int.abs_eq_natAbs, norm_zpow_abs]
+
+@[to_additive norm_isUnit_zsmul]
+theorem norm_zpow_isUnit (a : E) {n : ℤ} (hn : IsUnit n) : ‖a ^ n‖ = ‖a‖ := by
+  rw [← norm_pow_natAbs, Int.isUnit_iff_natAbs_eq.mp hn, pow_one]
+
+@[simp]
+theorem norm_units_zsmul {E : Type*} [SeminormedAddGroup E] (n : ℤˣ) (a : E) : ‖n • a‖ = ‖a‖ :=
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Analysis/Normed/Group/Basic.lean:391]: @eric-wieser Which variable is repeated? `E` is a `SeminormedGroup`, not a `SeminormedAddGroup` in the context.

**Pull request closed without merging**